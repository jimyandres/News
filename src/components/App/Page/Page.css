.page {
  margin: 20px;
}

.interactions {
  display: flex;
  flex-direction: column;
  align-items: center;
}
