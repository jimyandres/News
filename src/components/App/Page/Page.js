import React from 'react';
import EnhancedTable from './EnhancedTableWithConditionalRendering';
import Search from '../../Search';

const Page = (props) => {
  const {
    // Search Component
    searchTerm,
    onChange,
    onSubmit,
    // EnhancedTable Component
    isError,
    error,
    list,
    onDismiss,
    isLoading,
    onClick,
  } = props;

  return (
    <div className="page">
      <div className="interactions">
        <Search
          value={searchTerm}
          onChange={onChange}
          onSubmit={onSubmit}
        >
          <span>Search </span>
        </Search>
      </div>
      <EnhancedTable
        isError={isError}
        error={error}
        list={list}
        onDismiss={onDismiss}
        isLoading={isLoading}
        onClick={onClick}
        buttonText="Try Again"
      />
    </div>
  );
}

export default Page;

export {
  Search,
};
