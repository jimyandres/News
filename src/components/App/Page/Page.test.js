import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import Page from './Page';

configure({ adapter: new Adapter() });

describe('Page', () => {

	const props = {
		value: 'redux',
		onChange: jest.fn(),
		onSubmit: jest.fn(),
		isError: false,
		error: {},
		list: [],
		onDismiss: jest.fn(),
		isLoading: false,
		onClick: jest.fn(),
	};

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Page {...props}/>, div);
	});

	test('has a valid snapshot', () => {
		const component = shallow(<Page />);
		expect(shallowToJson(component)).toMatchSnapshot();
	});
});
