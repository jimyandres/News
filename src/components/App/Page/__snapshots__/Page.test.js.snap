// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page has a valid snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <Search
      onChange={[Function]}
      onSubmit={[Function]}
      value=""
    >
      <span>
        Search 
      </span>
    </Search>
  </div>
  <Component
    buttonText="Try Again"
    error={null}
    isError={false}
    isLoading={false}
    list={null}
    onClick={[Function]}
    onDismiss={[Function]}
  />
</div>
`;
